### PR TITLE
refactor(connector): Move connector Dlocal and Square from router to hyperswitch_connector crate

### PR DESCRIPTION
**Pull request by AnuthaDev**
## Type of Change

- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement
- [x] Refactoring
- [ ] Dependency updates
- [ ] Documentation
- [ ] CI/CD

## Description
Move conector Dlocal and Square from router to hyperswitch_connector crate

### Additional Changes

- [ ] This PR modifies the API contract
- [ ] This PR modifies the database schema
- [ ] This PR modifies application configuration/environment variables

## How did you test it?
1. Create Merchant Account
```
curl --location 'http://localhost:8080/accounts' \
--header 'Content-Type: application/json' \
--header 'Accept: application/json' \
--header 'api-key: test_admin' \
--data-raw '{
  "merchant_id": "merchant_1727684785",
  "locker_id": "m0010",
  "merchant_name": "NewAge Retailer",
  "merchant_details": {
    "primary_contact_person": "John Test",
    "primary_email": "JohnTest@test.com",
    "primary_phone": "sunt laborum",
    "secondary_contact_person": "John Test2",
    "secondary_email": "JohnTest2@test.com",
    "secondary_phone": "cillum do dolor id",
    "website": "https://www.example.com",
    "about_business": "Online Retail with a wide selection of organic products for North America",
    "address": {
      "line1": "1467",
      "line2": "Harrison Street",
      "line3": "Harrison Street",
      "city": "San Fransico",
      "state": "California",
      "zip": "94122",
      "country": "US"
    }
  },
  "return_url": "https://google.com/success",
  "webhook_details": {
    "webhook_version": "1.0.1",
    "webhook_username": "ekart_retail",
    "webhook_password": "password_ekart@123",
    "payment_created_enabled": true,
    "payment_succeeded_enabled": true,
    "payment_failed_enabled": true
  },
  "sub_merchants_enabled": false,
  "metadata": {
    "city": "NY",
    "unit": "245"
  },
  "primary_business_details": [
    {
      "country": "US",
      "business": "food"
    }
  ]
}'
```

2. Create API Key
```
curl --location 'http://localhost:8080/api_keys/merchant_1727684450' \
--header 'Content-Type: application/json' \
--header 'Accept: application/json' \
--header 'api-key: test_admin' \
--data '{
  "name": "API Key 1",
  "description": null,
  "expiration": "2038-01-19T03:14:08.000Z"
}'
```

3. Create Payment Connector for Dlocal
```
curl --location 'http://localhost:8080/account/merchant_1727684450/connectors' \
--header 'Content-Type: application/json' \
--header 'Accept: application/json' \
--header 'api-key: test_admin' \
--data '{
    "connector_type": "payment_processor",
     "connector_name": "dlocal",
    "connector_account_details": {
        "auth_type": "SignatureKey",
        "api_key": "_____",
        "key1": "_____",
        "api_secret" : "_____"

    },
    "test_mode": false,
    "disabled": false,
     "payment_methods_enabled": [
        {
            "payment_method": "card",
            "payment_method_types": [
                {
                    "payment_method_type": "credit",
                    "card_networks": [
                        "Visa",
                        "Mastercard"
                    ],
                    "minimum_amount": 1,
                    "maximum_amount": 68607706,
                    "recurring_enabled": true,
                    "installment_payment_enabled": true
                },
                {
                    "payment_method_type": "debit",
                    "card_networks": [
                        "Visa",
                        "Mastercard"
                    ],
                    "minimum_amount": 1,
                    "maximum_amount": 68607706,
                    "recurring_enabled": true,
                    "installment_payment_enabled": true
                }
            ]
        }
    ],
    "metadata": {
        "city": "NY",
        "unit": "245",
        "brand_id": "001",
        "destination_account_number": "5001000001223369"
    },
    "business_country": "US",
    "business_label": "food"
}'
```

4. Payment Create
```
curl --location 'http://localhost:8080/payments' \
--header 'Content-Type: application/json' \
--header 'Accept: application/json' \
--header 'api-key: ______' \
--data-raw '{
    "amount": 60000,
    "currency": "BRL",
    "confirm": true,
    "capture_method": "automatic",
    "capture_on": "2024-09-10T10:11:12Z",
    "amount_to_capture": 60000,
    "customer_id": "jsdhfgbds",
    "email": "guest@example.com",
    "name": "John Doe",
    "phone": "999999999",
    "phone_country_code": "+1",
    "description": "Its my first payment request",
    "authentication_type": "no_three_ds",
    "return_url": "https://duck.com",
    "payment_method": "card",
    "payment_method_type": "debit",
    "payment_method_data": {
        "card": {
            "card_number": "4242424242424242",
            "card_exp_month": "02",
            "card_exp_year": "37",
            "card_holder_name": "joseph Doe",
            "card_cvc": "999"
        }
    },
    "billing": {
        "address": {
            "line1": "1467",
            "line2": "Harrison Street",
            "line3": "Harrison Street",
            "city": "San Fransico",
            "state": "California",
            "zip": "94122",
            "country": "BR",
            "first_name": "joseph",
            "last_name": "Doe"
        },
        "phone": {
            "number": "8056594427",
            "country_code": "+91"
        }
    },
    "shipping": {
        "address": {
            "line1": "1467",
            "line2": "Harrison Street",
            "line3": "Harrison Street",
            "city": "San Fransico",
            "state": "California",
            "zip": "94122",
            "country": "BR",
            "first_name": "joseph",
            "last_name": "Doe"
        },
        "phone": {
            "number": "8056594427",
            "country_code": "+91"
        }
    },
    "statement_descriptor_name": "joseph",
    "statement_descriptor_suffix": "JS",
    "metadata": {
        "udf1": "value1",
        "new_customer": "true",
        "login_date": "2019-09-10T10:11:12Z"
    },
    "routing": {
        "type": "single",
        "data": "dlocal"
    }
}'
```

5. Response
```
{
    "payment_id": "pay_2cRAibQolK8jdZr1Teai",
    "merchant_id": "merchant_1727684450",
    "status": "succeeded",
    "amount": 60000,
    "net_amount": 60000,
    "amount_capturable": 0,
    "amount_received": 60000,
    "connector": "dlocal",
    "client_secret": "pay_2cRAibQolK8jdZr1Teai_secret_tINYaNt6LoYfuq2wD2Am",
    "created": "2024-09-30T08:20:54.759Z",
    "currency": "BRL",
    "customer_id": "jsdhfgbds",
    "customer": {
        "id": "jsdhfgbds",
        "name": "John Doe",
        "email": "guest@example.com",
        "phone": "999999999",
        "phone_country_code": "+1"
    },
    "description": "Its my first payment request",
    "refunds": null,
    "disputes": null,
    "mandate_id": null,
    "mandate_data": null,
    "setup_future_usage": null,
    "off_session": null,
    "capture_on": null,
    "capture_method": "automatic",
    "payment_method": "card",
    "payment_method_data": {
        "card": {
            "last4": "4242",
            "card_type": null,
            "card_network": null,
            "card_issuer": null,
            "card_issuing_country": null,
            "card_isin": "424242",
            "card_extended_bin": null,
            "card_exp_month": "02",
            "card_exp_year": "37",
            "card_holder_name": null,
            "payment_checks": null,
            "authentication_data": null
        },
        "billing": null
    },
    "payment_token": null,
    "shipping": {
        "address": {
            "city": "San Fransico",
            "country": "BR",
            "line1": "1467",
            "line2": "Harrison Street",
            "line3": "Harrison Street",
            "zip": "94122",
            "state": "California",
            "first_name": "joseph",
            "last_name": "Doe"
        },
        "phone": {
            "number": "8056594427",
            "country_code": "+91"
        },
        "email": null
    },
    "billing": {
        "address": {
            "city": "San Fransico",
            "country": "BR",
            "line1": "1467",
            "line2": "Harrison Street",
            "line3": "Harrison Street",
            "zip": "94122",
            "state": "California",
            "first_name": "joseph",
            "last_name": "Doe"
        },
        "phone": {
            "number": "8056594427",
            "country_code": "+91"
        },
        "email": null
    },
    "order_details": null,
    "email": "guest@example.com",
    "name": "John Doe",
    "phone": "999999999",
    "return_url": "https://duck.com/",
    "authentication_type": "no_three_ds",
    "statement_descriptor_name": "joseph",
    "statement_descriptor_suffix": "JS",
    "next_action": null,
    "cancellation_reason": null,
    "error_code": null,
    "error_message": null,
    "unified_code": null,
    "unified_message": null,
    "payment_experience": null,
    "payment_method_type": "debit",
    "connector_label": null,
    "business_country": null,
    "business_label": "default",
    "business_sub_label": null,
    "allowed_payment_method_types": null,
    "ephemeral_key": {
        "customer_id": "jsdhfgbds",
        "created_at": 1727684454,
        "expires": 1727688054,
        "secret": "epk_ed5899814cc94112940d95f020cdb9aa"
    },
    "manual_retry_allowed": false,
    "connector_transaction_id": "T-413100-f19a6e86-e95d-4e9e-bb5c-e27371ae8aa0",
    "frm_message": null,
    "metadata": {
        "udf1": "value1",
        "login_date": "2019-09-10T10:11:12Z",
        "new_customer": "true"
    },
    "connector_metadata": null,
    "feature_metadata": null,
    "reference_id": "pay_2cRAibQolK8jdZr1Teai_1",
    "payment_link": null,
    "profile_id": "pro_9mhY0ulaJuedDbQRGBDi",
    "surcharge_details": null,
    "attempt_count": 1,
    "merchant_decision": null,
    "merchant_connector_id": "mca_x0MV7McpNzabX1vANWza",
    "incremental_authorization_allowed": null,
    "authorization_count": null,
    "incremental_authorizations": null,
    "external_authentication_details": null,
    "external_3ds_authentication_attempted": false,
    "expires_on": "2024-09-30T08:35:54.759Z",
    "fingerprint": null,
    "browser_info": null,
    "payment_method_id": null,
    "payment_method_status": null,
    "updated": "2024-09-30T08:20:56.357Z",
    "charges": null,
    "frm_metadata": null,
    "merchant_order_reference_id": null,
    "order_tax_amount": null,
    "connector_mandate_id": null
}
```

## Checklist

- [x] I formatted the code `cargo +nightly fmt --all`
- [x] I addressed lints thrown by `cargo clippy`
- [x] I reviewed the submitted code
- [ ] I added unit tests for my changes where possible

### CODE REVIEW

**Review by deepanshu-iiitu — APPROVED** [review]

**Review by deepanshu-iiitu — DISMISSED** [review]

File: crates/hyperswitch_connectors/src/connectors/square.rs [hunk 1]
```diff
@@ -2,33 +2,78 @@ pub mod transformers;
...
+    configs::Connectors,
+    consts, errors,
+    events::connector_api_logs::ConnectorEvent,
+    types::{self, PaymentsAuthorizeType, Response},
+    webhooks::{IncomingWebhook, IncomingWebhookRequestDetails},
+};
+use masking::{Mask, Maskable, PeekInterface};
+// use common_utils::{ext_traits::ByteSliceExt, request::RequestContent};
+// use error_stack::ResultExt;
+// use masking::PeekInterface;
```
Comment (SanchithHegde) [inline crates/hyperswitch_connectors/src/connectors/square.rs:47]: Can we remove this commented code?

File: crates/hyperswitch_connectors/src/connectors/square.rs [hunk 2]
```diff
@@ -2,33 +2,78 @@ pub mod transformers;
...
+//         request::{self, Mask},
+//         ConnectorIntegration, ConnectorValidation,
+//     },
+//     types::{
+//         self,
+//         api::{self, ConnectorCommon, ConnectorCommonExt},
+//         ErrorResponse, Response,
+//     },
+//     utils::BytesExt,
+// };
```
Comment (SanchithHegde) [inline crates/hyperswitch_connectors/src/connectors/square.rs:75]: This one as well.

File: crates/hyperswitch_connectors/src/connectors/square/transformers.rs [hunk 3]
```diff
@@ -1,38 +1,55 @@
+// use error_stack::ResultExt;
+// use masking::{ExposeInterface, PeekInterface, Secret};
+// use serde::{Deserialize, Serialize};
+
+// use crate::{
+//     connector::utils::{self, CardData, PaymentsAuthorizeRequestData, RouterData},
+//     core::errors,
+//     types::{self, api, domain, storage::enums},
+//     unimplemented_payment_method,
+// };
```
Comment (SanchithHegde) [inline crates/hyperswitch_connectors/src/connectors/square/transformers.rs:10]: This one as well.

**Review by SanchithHegde — COMMENTED** [review]
Other than that, looks good to me!

File: crates/hyperswitch_connectors/src/connectors/square.rs (hunk 1 above)
Comment (AnuthaDev) [reply crates/hyperswitch_connectors/src/connectors/square.rs]: Done

**Review by deepanshu-iiitu — APPROVED** [review]

**Review by sahkal — APPROVED** [review]

**Pull request merged**